app/integrations.go

```diff
@@ -26,6 +26,21 @@ type ListedApp struct {
 	Enabled   bool `json:"enabled"`
 }
 
+func (a *App) CheckFreemiumLimitsForConfigSave(oldConfig, newConfig *model.Config) *model.AppError {
+	if !oldConfig.FeatureFlags.CloudFree {
+		return nil
+	}
+
+	for pluginId, newState := range newConfig.PluginSettings.PluginStates {
+		oldState, ok := oldConfig.PluginSettings.PluginStates[pluginId]
+		if newState.Enable && !(ok && oldState.Enable) {
+			return a.checkIfIntegrationMeetsFreemiumLimits(pluginId)
```

Comment:
I think we only want to return here if it is an `error`. Otherwise, if there are multiple plugins that would newly be enabled, we will end up enabling them all even if only the first one should be eligibile for enabling.

Reply:
Nevermind, was reading an earlier draft instead of the latest diff. Looks good here now.

Reply:
Double nevermind: I think my first comment still stands.